src/Client.ts

```diff
@@ -312,14 +312,18 @@ export default class Client {
     }
     const timestamp = options?.timestamp || new Date()
     const msg = await this.encodeMessage(recipient, timestamp, content, options)
+    const msgBytes = msg.toBytes()
+
     await Promise.all(
       topics.map(async (topic) => {
-        const wakuMsg = await WakuMessage.fromBytes(msg.toBytes(), topic, {
+        const wakuMsg = await WakuMessage.fromBytes(msgBytes, topic, {
           timestamp,
         })
         return this.sendWakuMessage(wakuMsg)
       })
     )
+
+    return this.decodeMessage(msgBytes, topics[0])
```

Comment:
I wish I didn't have to do this, but we only include things like the `contentType` and the decrypted content in the `decodeMessage` function. The Message only stores the resulting bytes. I'd rather not copy all the code for deriving that into a second place for maintainability's sake.

We should look at refactoring the message encoding/decoding so that this logic lives inside the message instead of the client.

Reply:
Uff, I really dislike this. That's a lot of overhead for something that may or may not end up being used.